### PR TITLE
Sync protocol change

### PR DESCRIPTION
**Pull request by smtmfft**

### CODE REVIEW

File: lib/src/taiko/host.rs [hunk 1]
```diff
@@ -207,23 +207,26 @@ fn decode_blob_data(blob: &str) -> Vec<u8> {
...
         let segment = &origin_blob
-            [i * BLOB_FIELD_ELEMENT_BYTES + 1..i * BLOB_FIELD_ELEMENT_BYTES + 1 + segment_len];
+            [i * BLOB_FIELD_ELEMENT_BYTES + 1..i * BLOB_FIELD_ELEMENT_BYTES + segment_len];
         i += 1;
         decoded_len += segment_len;
         chunk.push(segment.to_vec());
     }
-    chunk.iter().flatten().cloned().collect()
+    let blob_vec: Vec<u8> = chunk.iter().flatten().cloned().collect();
+    assert!(decoded_len == expected_len && blob_vec.len() < expected_len);
```
Comment (johntaiko) [inline lib/src/taiko/host.rs:228]: if expected_len == 0 then block_vec.len() == expected_len

File: lib/src/taiko/host.rs [hunk 2]
```diff
@@ -207,23 +207,26 @@ fn decode_blob_data(blob: &str) -> Vec<u8> {
...
     // the first 32 bytes is the length of the blob
     // every first 1 byte is reserved.
-    assert!(expected_len <= (BLOB_FIELD_ELEMENT_NUM - 1) * (BLOB_FIELD_ELEMENT_BYTES - 1));
+    // assert!(expected_len <= (BLOB_FIELD_ELEMENT_NUM - 1) * (BLOB_FIELD_ELEMENT_BYTES - 1));
+    assert!(expected_len <= BLOB_FIELD_ELEMENT_NUM * BLOB_FIELD_ELEMENT_BYTES);
     let mut chunk: Vec<Vec<u8>> = Vec::new();
-    let mut decoded_len = 0;
+    let mut decoded_len = 32;
     let mut i = 1;
     while decoded_len < expected_len && i < BLOB_FIELD_ELEMENT_NUM {
```
Comment (johntaiko) [inline lib/src/taiko/host.rs:215]: decode_len begins from 32, does expected_len begins from 0 ?

File: lib/src/taiko/host.rs (hunk 1 above)
Comment (smtmfft) [reply lib/src/taiko/host.rs]: <= 32 means a error.... because of the design, anyway, this one is deprecated, let's move to OP's impl.

**Comment by smtmfft** [comment]
Close for enc/dec upgrading.

**Pull request closed without merging**